pyo3-build-config/src/impl_.rs

```diff
@@ -1327,17 +1327,19 @@ fn is_cpython_lib_dir(path: &str, v: &Option<PythonVersion>) -> bool {
 }
 
 /// recursive search for _sysconfigdata, returns all possibilities of sysconfigdata paths
-fn search_lib_dir(path: impl AsRef<Path>, cross: &CrossCompileConfig) -> Vec<PathBuf> {
+fn search_lib_dir(path: impl AsRef<Path>, cross: &CrossCompileConfig) -> Result<Vec<PathBuf>> {
     let mut sysconfig_paths = vec![];
-    for f in fs::read_dir(path).expect("Path does not exist") {
+    for f in fs::read_dir(path)
+        .context("failed to search the lib dir, please check your path and permissions.")?
```

Comment:
I think we should name the relevant environment variables here or possibly add more targeted calls to `context` at the call sites of `search_lib_dir` so the user knows what to check exactly.

Reply:
Added a reference to `PYO3_CROSS_LIB_DIR`